server/sockets/join.js

```diff
@@ -19,16 +20,28 @@ const {
  *
  * @param {Server} io
  */
+
 const matchMaker = async (io) => {
   while (getWaitingUserLen() > 1) {
-    const chat = await createChat(getRandomPairFromWaitingList());
+    const [user1, user2] = getRandomPairFromWaitingList();
+    
+    // Check if either user is blocked
+    if (
+        await isUserBlocked(user1.loginId, user2.loginId) 
+        || 
+        await isUserBlocked(user2.loginId, user1.loginId)) {
+      continue
+    }
```

Comment:
You can implement this in isUserBlocked itself, that way you don't need this condition, and it can just be one.

And instead of the destructing, we could just still keep this code as it is and just pass the users to isUserBlocked.